src/System.Binary.Base64/System/Binary/Base64Decoder.cs

```diff
@@ -207,39 +207,79 @@ public TransformationStatus Transform(ReadOnlySpan<byte> source, Span<byte> dest
         /// <returns>Number of bytes written to the buffer.</returns>
         public static int DecodeInPlace(Span<byte> buffer)
         {
-            // TODO: Fix
-            int di = 0;
-            int si = 0;
-            //byte r0, r1, r2;
-            int padding = 0;
+            ref byte bufferBytes = ref buffer.DangerousGetPinnableReference();
 
-            if (buffer[buffer.Length - 1] == s_encodingPad)
+            int bufferLength = buffer.Length & ~0x3;  // only decode input up to the closest multiple of 4.
+
+            int sourceIndex = 0;
+            int destIndex = 0;
+
+            if (buffer.Length < 4) return 0;
```

Comment:
Hmm... bufferLength is buffer.Length /4 which rounds down. So if buffer.Length is < 4, bufferLength would be 0 and I could just instead change the check to that. However, I will wait until we decide on the return value of the method. We would need to have the state "NeedMoreExit", for buffer lengths between 1 and 3 (inclusive), so I will wait for that before making this change.